src/glob.zig

```diff
@@ -0,0 +1,3021 @@
+        GlobWalker.convertUtf8ToCodepoints(
+            codepoints,
+            this.pattern[pattern_component.start .. pattern_component.start + pattern_component.len],
+        );
+        pattern_component.unicode_set = true;
+        return codepoints;
+    }
+
+    fn appendMatchedPath(
+        this: *GlobWalker,
+        entry_name: []const u8,
+        dir_name: [:0]const u8,
+    ) !void {
+        const subdir_parts: []const []const u8 = &[_][]const u8{
+            dir_name[0..dir_name.len],
+            entry_name,
+        };
+        const name = try this.join(subdir_parts);
+        try this.matchedPaths.append(this.arena.allocator(), BunString.fromBytes(name));
+    }
+
+    fn appendMatchedPathSymlink(this: *GlobWalker, symlink_full_path: []const u8) !void {
+        const name = try this.arena.allocator().dupe(u8, symlink_full_path);
+        try this.matchedPaths.append(this.arena.allocator(), BunString.fromBytes(name));
+    }
+
+    inline fn join(this: *GlobWalker, subdir_parts: []const []const u8) ![]u8 {
+        return if (!this.absolute)
+            // If relative paths enabled, stdlib join is preferred over
+            // ResolvePath.joinBuf because it doesn't try to normalize the path
+            try std.fs.path.join(this.arena.allocator(), subdir_parts)
+        else
+            try this.arena.allocator().dupe(u8, ResolvePath.join(subdir_parts, .auto));
+    }
+
+    inline fn startsWithDot(filepath: []const u8) bool {
+        if (comptime !isWindows) {
+            return filepath[0] == '.';
+        } else {
+            return std.unicode.utf16DecodeSurrogatePair(&filepath[0..2]) == '.' catch false;
+        }
+    }
+
+    fn hasLeadingDot(filepath: []const u8, comptime allow_non_utf8: bool) bool {
+        if (comptime bun.Environment.isWindows and allow_non_utf8) {
+            // utf-16
+            if (filepath.len >= 4 and filepath[1] == '.' and filepath[3] == '/')
+                return true;
+        } else {
+            if (filepath.len >= 2 and filepath[0] == '.' and filepath[1] == '/')
+                return true;
+        }
+
+        return false;
+    }
+
+    /// NOTE This doesn't check that there is leading dot, use `hasLeadingDot()` to do that
+    fn removeLeadingDot(filepath: []const u8, comptime allow_non_utf8: bool) []const u8 {
+        if (comptime bun.Environment.isWindows and allow_non_utf8) {
+            return filepath[4..];
+        } else {
+            return filepath[2..];
+        }
+    }
+
+    fn checkSpecialSyntax(pattern: []const u8) bool {
+        if (pattern.len < 16) {
+            for (pattern[0..]) |c| {
+                switch (c) {
+                    '*', '[', '{', '?', '!' => return true,
+                    else => {},
+                }
+            }
+            return false;
+        }
+
+        const syntax_tokens = comptime [_]u8{ '*', '[', '{', '?', '!' };
+        const needles: [syntax_tokens.len]@Vector(16, u8) = needles: {
```

Comment:
can this be `comptime`?
```suggestion
        const needles: [syntax_tokens.len]@Vector(16, u8) = comptime needles: {
```